lib/matplotlib/axes/_base.py

```diff
@@ -4322,13 +4321,14 @@ def get_tightbbox(self, renderer, call_axes_locator=True,
         else:
             self.apply_aspect()
 
-        bb_xaxis = self.xaxis.get_tightbbox(renderer)
-        if bb_xaxis:
-            bb.append(bb_xaxis)
+        if self.axison:
```

Comment:
This change (self.axison), caused a few tests top change.  So this is probably a breaking change.  OTOH, I think its correct - it doesn't make sense to include the axis in the tight layout if its been turned off.